astropy/coordinates/sky_coordinate.py

```diff
@@ -1575,7 +1581,24 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
             gr = location.gravitational_redshift(obstime)
             # barycentric redshift according to eq 28 in Wright & Eastmann (2014),
             # neglecting Shapiro delay and effects of the star's own motion
-            zb = gamma_obs * (1 + targcart.dot(beta_obs)) / (1 + gr/speed_of_light) - 1
+            zb = gamma_obs * (1 + targcart.dot(beta_obs)) / (1 + gr/speed_of_light)
+            # try and get terms corresponding to stellar motion.
```

Comment:
I think rather than the `try/except`, I would just do:
```
if ierc_cast.differentials:
    try:
        beta_star = ...
    except u.UnitConversionError:
        warnings...
```
This mostly as then the intent is clearer (a comment is no longer needed), but also as it would actually fail on a (so far completely hypothetical) case where there are accelerations ('s2') but no velocities. Plus, having no velocities is likely still the most common case so better to not let that suffer in speed (except part of a try/except is the more costly one).

Reply:
Good idea! It also avoids costly access to the cartesian of the epoch-level icrs coordinates if it's not needed.